### ISSUE TITLE
`landmark-roles` rule: Allow the `<aside>` element to be contained within other landmarks

### ISSUE THREAD

**Issue by SaekiTominaga** [markuplint/markuplint#3280]
`landmark-roles` ルールでは banner, main, complementary, contentinfo の4つのロールに対してトップレベルであるかどうかのチェックが行われています。そのうえで `<aside>` 要素が `role="complementary"` に紐付けられているので、ほかのランドマーク内に入れた場合に warning となります。（v4.14.1 と v5.0.0-alpha.0 で確認）

```html
<main>
	<aside>text</aside>
</main>
```

しかし2022年に `<aside>` 要素のアクセシビリティマッピングが変更され、状況に応じて `complementary` / `generic` となったことで、ほかのランドマーク内に含めても問題なくなりました。その後ブラウザやツールも徐々に対応が進んでいます。

ブラウザは2023〜2024年にかけて対応が行われています。👉[`<aside>` 要素のアクセシビリティマッピングの変更が各ブラウザーに反映されている · Issue #468 · SaekiTominaga/blog.w0s.jp](https://github.com/SaekiTominaga/blog.w0s.jp/issues/468)

また他のツールの状況として、Axe は以下の対応が行われています。

- 2021年に `<main>` 要素内の `<aside>` 要素の警告を取り止め 👉[axe-core 4.2](https://github.com/dequelabs/axe-core/milestone/19?closed=1)
- 2026年1月に `landmark-complementary-is-top-level` ルールそのものを廃止した（まだリリースはされていない） 👉[landmark-complementary-is-top-level has been obsoleted by recent ARIA and HTML-AAM changes · Issue #4950 · dequelabs/axe-core](https://github.com/dequelabs/axe-core/issues/4950)

ということで、`<aside>` 要素の配置位置はそろそろ本来の HTML 仕様に準じたもの（トップレベル以外を許容）にしてよい頃合いかと思います。
